lib/push-jenkins-update.js

```diff
@@ -118,6 +132,25 @@ function createGhStatus (options, logger, cb) {
   })
 }
 
+function findExistingCiComment (options, logger, cb) {
+  async.parallel([
+    botUsername.resolve,
+    githubClient.issues.getComments.bind(githubClient.issues, {
```

Comment:
This will only return one page (i.e. 30), we have issues (https://github.com/nodejs/node/pull/26322) with much more, so this won't find the comment and revert to the current behavior...

I'm playing with the graphQL API, that can return 100. I was thinking of fetching a batch from the top, and from the bottom.

BTW: the graphQL schema has `viewerDidAuthor` property, so no need for the bot's id anyway.

Reply:
Good catch, obviously didn't remember that at first. Fetching all pages of entries is something we'd have to do elsewhere, e.g. in [./lib/node-repo.js](https://github.com/nodejs/github-bot/blob/master/lib/node-repo.js#L138).

> I'm playing with the graphQL API, that can return 100. I was thinking of fetching a batch from the top, and from the bottom.

Cool, the GraphQL API is something I've been wanting to try out for a while too!

Reply:
I'll try to push a suggestion, as soon as I refresh my sinon & supertest skills...